### ISSUE TITLE
Streaming -PassThru output in v4

### ISSUE THREAD

**Issue by redoz** [pester/Pester#550]
I'm not sure if reasonable/possible but it would be great if the -PassThru data could be streamed out rather than returned at the end.  My test suite takes several minutes to run and it would be nice if I could get structured data to update my UI as the tests are executed rather than as a large chunk at the end.

**Comment by dlwyatt** [comment]
On the -PassThru object, no.  But maybe we can add something similar to event handlers; you pass in a script block to run for each Describe / Context / It, that sort of thing.

**Comment by nohwnd** [comment]
I tried to do that (callbacks) a while back in hope for a better integration with VS rest runner. Matbe I still have the code somewhere.

**Comment by redoz** [comment]
@dlwyatt yeah that would work, hopefully it can be designed in such a way it's easy to drop in different consumers, maybe even multiple ones?

Will v4 maintain backwards compatibility with PS2?

**Comment by dlwyatt** [comment]
I'm not 100% sure on that one.  We get held back quite a bit on new features by having to support PSv2, and at this point, the only operating systems that can even run PSv2 have already exited mainstream support (though extended support goes until 2020, if I remember correctly.)

**Comment by redoz** [comment]
Well, maybe more importantly, is there any OS that will _only_ run PSv2? Didn't MS do a pretty good job of backporting v3/4/5 to all supported OSs?

**Comment by dlwyatt** [comment]
the last OS that couldn't be upgraded past PSv2 was XP / 2003.  Other than maybe some XP embedded stuff, that's all past extended support now.

However, I've been in those big enterprises that won't let you upgrade stuff, and I wouldn't be surprised at all to find many admins / companies out there that still have to put up with v2 for now.  Most of the time, it doesn't hurt to make modules v2-compatible, but we've run into problems several times so far (where we'd want to take advantage of the AST, but can't.)

Personally, I'm leaning toward making v3 or v4 the last version of Pester to support PowerShell v2, but we'll see how it goes.  Will probably put that out as an RFC or poll or something.

**Comment by redoz** [comment]
Well I don't work in in a large enterprise so I'm all for keeping v3 the latest version supporting PSv2, then v4 can hopefully move faster and further.

**Comment by it-praktyk** [comment]
What is a conclusion? 
Is it related to #447?

**Comment by nohwnd** [comment]
@it-praktyk Let's keep this open. I am planning to change the internals to allow extensions, and this could be one of them.

**Comment by nohwnd** [comment]
This was a v4-era request. Streaming results as tests run is now possible via the Pester 5 plugin hooks (EachTest callbacks), which fire during the run rather than handing you one big -PassThru object at the end. That covers the 'update my UI as tests execute' need. Closing as the underlying capability exists now and the v4 shape is obsolete.
